### PR TITLE
Mar 40 filter endpoints

### PR DESCRIPTION
**Pull request by abiolafasanya**
you can check this one that was used for the previous cohort

# Fixes Issue/Linear Ticket

> This could be an existing issue or a linear ticket
>
> - Github Issue Example: My PR Closes #{ISSUE}
> - Linear Ticket Example: Fixes ID-#{ISSUE}

# Changes proposed

> Fixed product card type error
> Fetched Data for Navigation

# Check List (Check all the applicable boxes)

:rotating_light:Please review the [style guide for contributing](add link here) and [guidelines for contributing](add link here) to this repository.

- [x] My code follows the code style of this project.
- [x] This PR does not contain plagiarized content.
- [x] The title and description of the PR is clear and explains the approach.
- [x] I am making a pull request against the **main branch** (left side).
- [x] My commit messages styles matches our requested structure.
- [x] My code additions will fail neither code linting checks nor unit test.
- [x] I am only making changes to files I was requested to.

# Screenshots/ Videos

### CODE REVIEW

File: modules/marketplace/component/CategoriesNav/ButtonCat.tsx [hunk 1]
```diff
@@ -49,12 +51,12 @@ const ButtonCat = ({ category, index, handleActiveNav, active }: CategoriesProps
         onMouseOver={() => setPopupClass(true)}
         onMouseLeave={() => setPopupClass(false)}
       >
-        {category.subcategories.map((item: { name: string }, i: number) => (
```
Comment (billmal071) [inline modules/marketplace/component/CategoriesNav/ButtonCat.tsx:52]: why did you remove the typing?

File: modules/marketplace/component/CategoriesNav/CategoriesNav.tsx [hunk 2]
```diff
@@ -24,6 +26,7 @@ const CategoriesNav = (props: CategoriesNavProps) => {
 
   const { authenticated } = useAuthentication();
   const { navItems } = props;
+  console.log(navItems)
```
Comment (billmal071) [inline modules/marketplace/component/CategoriesNav/CategoriesNav.tsx]: Hmm...

File: modules/marketplace/component/CategoriesNav/CategoriesNav.tsx [hunk 3]
```diff
@@ -81,13 +84,13 @@ const CategoriesNav = (props: CategoriesNavProps) => {
                 <Link href={`/marketplace/wishlist`}>WishList</Link>
               </li>
             )}
-            {navItems.map((category, i: number) => {
+            {props.isLoading ? 'loading...' : navItems.map((category, i: number) => {
```
Comment (billmal071) [inline modules/marketplace/component/CategoriesNav/CategoriesNav.tsx]: What happened to using a loader?

**Review by billmal071 — CHANGES_REQUESTED** [review]

File: modules/marketplace/component/CategoriesNav/CategoriesNav.tsx (hunk 3 above)
Comment (abiolafasanya) [reply modules/marketplace/component/CategoriesNav/CategoriesNav.tsx]: It will be updated now

File: modules/marketplace/component/CategoriesNav/ButtonCat.tsx (hunk 1 above)
Comment (abiolafasanya) [reply modules/marketplace/component/CategoriesNav/ButtonCat.tsx:52]: the typing was Improved, when hovered on the type is
``` javascipt
(parameter) item: {
    name?: string | undefined;
}
```typescript

Below is the type supplying the above hover description

export type CategoryType = {
  name: string;
  subcategories: {name?: string}[];
};
``

File: modules/marketplace/component/CategoriesNav/CategoriesNav.tsx (hunk 2 above)
Comment (abiolafasanya) [reply modules/marketplace/component/CategoriesNav/CategoriesNav.tsx]: removed

**Review by Samadeen — APPROVED** [review]
